src/RoslynPad.Build/ExecutionHost.cs

```diff
@@ -706,9 +706,16 @@ async Task ReadReferences(CancellationToken cancellationToken)
 
                     cancellationToken.ThrowIfCancellationRequested();
 
+                    // Create a MetadataReferenceProperties with embedInteropTypes to true 
+                    // to integrate MS Office Interop types to the referencing compilation 
+                    // (only required for .Net Core)
+                    var metadataReferenceProperties = new MetadataReferenceProperties(embedInteropTypes: true);
+
                     MetadataReferences = references
                         .Where(r => !string.IsNullOrWhiteSpace(r))
-                        .Select(r => _roslynHost.CreateMetadataReference(r))
+                        .Select(r => Platform.IsCore && r.Contains(@"net20\Microsoft.Office.Interop") 
```

Comment:
I don't like that this solution is so specific to a version of office interop assemblies. A better option would be to add this to the `#r` directive, e.g.

```cs
#r "Intertop.dll, EmbedInteropTypes"
```